x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/allocation/TrainedModelAllocationNodeService.java

```diff
@@ -168,6 +178,20 @@ public void stop() {
 
     void loadQueuedModels() {
         TrainedModelDeploymentTask loadingTask;
+        if (latestState != null) {
+            List<String> unassignedIndices = AbstractJobPersistentTasksExecutor.verifyIndicesPrimaryShardsAreActive(
+                latestState,
+                expressionResolver,
+                // we allow missing as that means the index doesn't exist at all and our loading will fail for the models and we need
+                // to notify as necessary
+                true,
+                InferenceIndexConstants.INDEX_PATTERN
```

Comment:
```suggestion
                InferenceIndexConstants.INDEX_PATTERN, InferenceIndexConstants.nativeDefinitionStore(), 
```

`InferenceIndexConstants.INDEX_PATTERN` for the config and `InferenceIndexConstants.nativeDefinitionStore()` for the vocab and model definition.